src/include/api.h

```diff
@@ -96,11 +96,11 @@
         if ((config) != NULL)                                                               \
     WT_ERR(__wt_config_check((s), WT_CONFIG_REF(s, struct_name##_##func_name), (config), 0))
 
-#define API_END(s, ret)                                                                    \
+#define API_END_OK(s, ret, okerr)                                                          \
     if ((s) != NULL) {                                                                     \
         WT_TRACK_OP_END(s);                                                                \
         WT_SINGLE_THREAD_CHECK_STOP(s);                                                    \
-        if ((ret) != 0)                                                                    \
+        if ((ret) != 0 && (ret) != (okerr))                                                \
```

Comment:
I think this is to avoid the transaction being rolledbacked if we see an EBUSY return. In this case, it is the session drop call that receives the EBUSY. If there is no transaction running, we don't need to worry about this. If there is, I don't think this is a big deal either and we can let the transaction to rollback.

Reply:
According to our meeting, we no longer avoid transaction rollback.
To make it easier to see what is left I am going to resolve some conversations.